DiscordBot/Commands/EventCommands.cs

```
@@ -19,34 +19,107 @@ internal class EventCommands
-                message =>
-                    IsValidResponse(message.Content),
-                TimeSpan.FromSeconds(30));
+            var eventOperation = await GetUserResponse(context, interactivity, EventOperations);
+            if (eventOperation == null)
+            {
+                return;
+            }
+
+            switch (eventOperation)
+            {
+                case "create":
+                    await CreateEvent(context, interactivity);
+                    break;
+                case "rm":
+                    await context.RespondAsync("Not implemented yet!");
+                    break;
+                case "edit":
+                    await context.RespondAsync("Not implemented yet!");
+                    break;
+            }
+        }
+
+        public async Task CreateEvent(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the name of the event?");
+            var eventName = await GetUserResponse(context, interactivity);
+            if (eventName == null)
+            {
+                return;
+            }
+            
+            await context.RespondAsync($"{context.Member.Mention} - give an event description.");
+            var eventDescription = await GetUserResponse(context, interactivity);
+            if (eventDescription == null)
+            {
+                return;
+            }
 
-            if (response == null)
+            await context.RespondAsync($"{context.Member.Mention} - what time is your event?");
+            var eventTimeResponse = await GetUserResponse(context, interactivity);
+            if (eventTimeResponse == null)
             {
-                await context.RespondAsync($"{context.Member.Mention} - timed out");
+                return;
             }
-            else
+
+            if (!DateTimeOffset.TryParse(eventTimeResponse, out var eventTime))
```

I think it would be useful to get some more information about what was wrong with the date input, otherwise the user might get confused. Can we use `DateTime.Parse` (which should throw an exception if given invalid date) instead or will that return the wrong type for the DiscordEmbelBuilder? Or is there a version of the DateTimeOffset parse method which will throw an exception instead of returning a boolean? 🙂